IntegrationTests/tests_04_performance/test_01_resources/template/Sources/SwiftBootstrap/SwiftMain.swift

```diff
@@ -0,0 +1,164 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+import Foundation
+import AtomicCounter
+
+private final class SimpleHTTPServer: ChannelInboundHandler {
+    typealias InboundIn = HTTPServerRequestPart
+    typealias OutboundOut = HTTPServerResponsePart
+
+    private var files: [String] = Array()
+    private var seenEnd: Bool = false
+    private var sentEnd: Bool = false
+    private var isOpen: Bool = true
+
+    private let bodyLength = 100
+    private let numberOfAdditionalHeaders = 3
+
+    private var responseHead: HTTPResponseHead {
+        var head = HTTPResponseHead(version: HTTPVersion(major: 1, minor: 1), status: .ok)
+        head.headers.add(name: "Content-Length", value: "\(self.bodyLength)")
+        for i in 0..<self.numberOfAdditionalHeaders {
+            head.headers.add(name: "X-Random-Extra-Header", value: "\(i)")
+        }
+        return head
+    }
+
+    private func responseBody(allocator: ByteBufferAllocator) -> ByteBuffer {
+        var buffer = allocator.buffer(capacity: self.bodyLength)
+        for i in 0..<self.bodyLength {
+            buffer.write(integer: UInt8(i % Int(UInt8.max)))
+        }
+        return buffer
+    }
+
+    public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+        if case .head(let req) = self.unwrapInboundIn(data), req.uri == "/allocation-test-1" {
+            ctx.write(self.wrapOutboundOut(.head(self.responseHead)), promise: nil)
+            ctx.write(self.wrapOutboundOut(.body(.byteBuffer(self.responseBody(allocator: ctx.channel.allocator)))), promise: nil)
+            ctx.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
+        }
+    }
+}
+
+@_cdecl("swift_main")
+public func swiftMain() -> Int {
+    final class RepeatedRequests: ChannelInboundHandler {
+        typealias InboundIn = HTTPClientResponsePart
+        typealias OutboundOut = HTTPClientRequestPart
+
+        private let numberOfRequests: Int
+        private var remainingNumberOfRequests: Int
+        private var doneRequests = 0
```

Comment:
Nit: this can presumably be a computed variable from `numberOfRequests - remainingNumberOfRequests`, right?